core/src/main/scala/io/finch/ToService.scala

```diff
@@ -107,9 +110,16 @@ object ToService {
 
             val accept =
               if (opts.negotiateContentType) req.accept.map(a => Accept.fromString(a)) else Nil
-            out.map(oa =>
-              conformHttp(oa.toResponse(ntrA(accept), ntrE(accept)), req.version, opts)
-            ).run
+
+            val promise = Promise[Output[A]]
+            val io = effect.runAsync(out) {
```

Comment:
Can we just do `effect.unsafeRunAsync` instead and avoid calling `unsafeRunSync` later?

Reply:
Sadly no, `unsafeRunAsync` is a method of `IO` only